common/scala/src/main/scala/whisk/core/entity/Exec.scala

```diff
@@ -93,9 +94,18 @@ protected[core] case class JavaExec(code: Attachment[String], main: String) exte
 
 protected[core] case class PythonExec(code: String) extends CodeExec[String](Exec.PYTHON)
 
-protected[core] case class BlackBoxExec(image: String) extends Exec(Exec.BLACKBOX) {
-    def size = image sizeInBytes
-    override val sentinelledLogs = false
+/**
+ * @param image the image name
+ * @param code an optional script or zip archive (as base64 encoded) string
+ */
+protected[core] case class BlackBoxExec(override val image: String, code: Option[String]) extends CodeExec[Option[String]](Exec.BLACKBOX) {
```

Comment:
There is no invariant that `(image == Exec.BLACKBOX_SKELETON) != code.isDefined`, right?

This means, I guess, that users can build their own skeleton and feed it a binary? Is this a { desired, tested, documented } feature?

I realize they may have to bypass the CLI to do that.